### PR TITLE
Reinforcement Learning Based Tractography

### PR DESCRIPTION
**Pull request by AntoineTheb**
Le son pour décrire ce papier:
https://www.youtube.com/watch?v=_asNhzXq72w

### CODE REVIEW

**Comment by AntoineTheb** [comment]
C'est la 300e PR ouverte sur ce repo ! :tada:

File: reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+cite:
+    authors: "Tingyi Wanyan, Lantao Liu, Eleftherios Garyfallidis"
+    title:   "Tractography using reinforcement learning and adaptive-expanding graphs"
+    venue:   "ISMRM 2015"
+pdf: "https://drive.google.com/open?id=1gYXIGYd6jRE0HCSvvx5NTRTBKUMLYBrb"
+---
+
+## Introduction
+
+Currently, three types of tractography algorithm exist: Deterministic, probabilistic and global algoritms. The authors propose a novel approach based on reinforcement-learning.
```
Comment (jhlegarreta) [inline reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md]: *(...) three types of tractography algorithm exist: Deterministic, (...)* to *(...) three types of tractography algorithms exist: deterministic, (...)*

File: reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+layout: review
+title: "Tractography using reinforcement learning and adaptive-expanding graphs"
+tags: reinforcement-learning tractography
```
Comment (jhlegarreta) [inline reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md]: The *reinforcement-learning* tag is not currently listed; we can either include it, modify the existing *reinforcement*, or else, use the latter in the review.

File: reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+author: "Antoine Théberge"
+cite:
+    authors: "Tingyi Wanyan, Lantao Liu, Eleftherios Garyfallidis"
+    title:   "Tractography using reinforcement learning and adaptive-expanding graphs"
+    venue:   "ISMRM 2015"
```
Comment (jhlegarreta) [inline reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md]: Does the paper date back to ISMRM 2015?

File: reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## Method
+
+![](/reinforcement-learning/images/tracto-td-learning/tree.png)
+
+The authors used the ISMRM challenge dataset which provided a starting and goal point for tractography. They then used a graph to represent their exploration and experiences. Each node represent a state in their search and has a value function, updated via Temporal-Difference (TD) learning:
+
+$$ V(s_t) \leftarrow V(s_t) + \alpha[R_{t+1} + \gamma V(s_{t+1}) - V(s_t)] $$
+
+where $$V(s_t)$$ is the value function, the sum of all expected rewards at this node, $$\alpha$$ is the learning rate, $$\gamma$$ a discount factor and $$R_{t+1}$$ is the reward at the next time step. It is worth noting that the value function is only updated when a valid path (ed. the definition of a valid path is not specified, probably a path that goes near or to the end goal) is found to prevent penalizing invalid paths, which are still useful for exploration. In the end, optimal tracts are determined by their total reward.
```
Comment (jodoin) [inline reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md]: do they mention what their reward function is?

File: reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md (hunk 1 above)
Comment (AntoineTheb) [reply reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md]: Fixed !

File: reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md (hunk 4 above)
Comment (AntoineTheb) [reply reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md]: Unfortunately not

File: reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md (hunk 3 above)
Comment (AntoineTheb) [reply reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md]: Woops, it was actually published at ISBI 2018

File: reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md (hunk 2 above)
Comment (AntoineTheb) [reply reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md]: Changed it for _reinforcement_

**Comment by jhlegarreta** [comment]
BTW, I ignore the rules of the conference, but we may need to check whether images can be reproduced.

**Comment by jodoin** [comment]
> BTW, I ignore the rules of the conference, but we may need to check whether images can be reproduced.

Good point.  In fact, since the paper is not on arxiv, I do not think that we can render public the poster and the paper.  Antoine, please check the copyright before we merge

**Comment by AntoineTheb** [comment]
@jodoin
https://www.ieee.org/content/dam/ieee-org/ieee/web/org/pubs/permissions_faq.pdf

We'd have to ask permission to use the images in this review, is it worth it ?

File: reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+where $$V(s_t)$$ is the value function, the sum of all expected rewards at this node, $$\alpha$$ is the learning rate, $$\gamma$$ a discount factor and $$R_{t+1}$$ is the reward at the next time step. It is worth noting that the value function is only updated when a valid path (ed. the definition of a valid path is not specified, probably a path that goes near or to the end goal) is found to prevent penalizing invalid paths, which are still useful for exploration. In the end, optimal tracts are determined by their total reward (ed. reward function not described).
+
+## Results
+
+(See poster, table 1 for results)
+
+We can see that the RL algorithm performs much better at finding valid tracts than a deterministic algorithm (EuDX, which is old and not used very much anymore).  However, we do not know many invalid tracts the RL algorithm has found. After all it might just brute-force it. It would have been much more informative to display precision, recall or any other metric than solely the true-positives rate. Also, to reiterate, they had the massive advantage of having an end goal, which most definitely helped them design their reward.
```
Comment (lemairecarl) [inline reinforcement-learning/_posts/2018-12-17-tracto_td_learning.md]: we do not know **how** many invalid tracts

**Comment by AntoineTheb** [comment]
@jodoin They're not, let's just close this.

**Pull request closed without merging**